dev/ci/semgrep-scan.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+set -e
+
+echo -e "--- :construction: cloning Semgrep rules\n"
+
+# clone the semgrep repo rules
+gh repo clone sourcegraph/security-semgrep-rules
+
+echo -e "--- :lock: running Semgrep scan\n"
```

Comment:
```suggestion
echo -e "--- :lock::semgrep: running Semgrep scan\n"
```
from https://github.com/buildkite/emojis/raw/main/img-buildkite-64/semgrep.png